pkg/playback/http_playback.go

```diff
@@ -0,0 +1,413 @@
+package playback
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+func forwardRequest(wrappedRequest *httpRequest, destinationURL string) (resp *http.Response, err error) {
+	client := &http.Client{
+		// set Timeout explicitly, otherwise the client will wait indefinitely for a response
+		Timeout: time.Second * 10,
+	}
+
+	// Create a identical copy of the request
+	req, err := http.NewRequest(wrappedRequest.Method, destinationURL, bytes.NewBuffer(wrappedRequest.Body))
+	if err != nil {
+		return nil, err
+	}
+	copyHTTPHeader(req.Header, wrappedRequest.Headers)
+
+	// Forward the request to the remote
+	res, err := client.Do(req)
+
+	if err != nil {
+		return nil, err
+	}
+	return res, nil
+}
+
+// Writes to the HTTP response using the given HTTP status code and error text as the response body. Simultaneously, prints the error text to stdout.
+// Note: since http.ResponseWriter streams the response, you will not be able to change the HTTP status code after calling this function.
+func writeErrorToHTTPResponse(w http.ResponseWriter, err error, statusCode int) {
+	w.WriteHeader(statusCode)
+	fmt.Fprintf(w, "%v\n", err)
```

Comment:
What do you think about writing the error as JSON to the application server, instead of plain text? All the other responses we are forwarding to the server (from Stripe) are JSON, so probably we want a vcr-related error to trigger the same error-handling flow as Stripe errors. If we just send plain text probably the user will get a JSON parse error and it will be harder for them to see the text of the error message.

Reply:
Ah good idea, that explains some weird coloring I was seeing. I updated it to return JSON in the same format as the normal Stripe API in 53bfbfdeca119d1b9e93ba6b0e0f5f7e0f4c6ae9.


Here's the before and after when using the Stripe CLI to send a request to the server that returns a `playback` error:

Before (with strange coloring):
<img width="639" alt="Screen Shot 2020-07-23 at 6 01 12 PM" src="https://user-images.githubusercontent.com/65417391/88343035-80a8be00-cd0e-11ea-8964-67cb16067312.png">

After (with proper coloring):
<img width="655" alt="Screen Shot 2020-07-23 at 6 00 11 PM" src="https://user-images.githubusercontent.com/65417391/88342942-5c4ce180-cd0e-11ea-9078-9dcae6061bcd.png">